store/sqlstore/channel_store.go

```diff
@@ -1178,10 +1148,7 @@ func (s SqlChannelStore) GetByNames(teamId string, names []string, allowFromCach
 		if _, err := s.GetReplica().Select(&dbChannels, query, props); err != nil && err != sql.ErrNoRows {
 			return nil, model.NewAppError("SqlChannelStore.GetByName", "store.sql_channel.get_by_name.existing.app_error", nil, "teamId="+teamId+", "+err.Error(), http.StatusInternalServerError)
 		}
-		for _, channel := range dbChannels {
-			channelByNameCache.AddWithExpiresInSecs(teamId+channel.Name, channel, CHANNEL_CACHE_SEC)
-			channels = append(channels, channel)
-		}
+		channels = append(channels, dbChannels...)
```

Comment:
the `channels` variable here is going to be empty always, so you don't need the concatenation.